src/main/java/com/flipkart/databuilderframework/engine/OptimizedMultiThreadedDataFlowExecutor.java

```diff
@@ -243,7 +243,12 @@ public DataContainer call() throws Exception {
                 try {
                     listener.beforeExecute(dataBuilderContext, dataFlowInstance, builderMeta, dataDelta, responseData);
                 } catch (Throwable t) {
-                    logger.error("Error running pre-execution execution listener: ", t);
+                    final String errorMessage = "Error running pre-execution execution listener: ";
+                    logger.error(errorMessage, t);
+                    if (listener.shouldThrowExceptionInBeforeExecute()) {
```

Comment:
Exact same code is getting repeated at three places. Create a method like, handleExceptionInBeforeExecute() in utils and use that at all the places.
Same can be done for afterExecute as well.